ethers-contract/ethers-contract-abigen/src/source/mod.rs

```diff
@@ -141,17 +133,22 @@ mod tests {
 
     #[test]
     fn parse_source() {
-        let rel = "../tests/solidity-contracts/console.json";
-        let abs = concat!(env!("CARGO_MANIFEST_DIR"), "/../tests/solidity-contracts/console.json");
-        let abs_url = concat!(
-            "file://",
-            env!("CARGO_MANIFEST_DIR"),
-            "/../tests/solidity-contracts/console.json"
-        );
-        let exp = Source::Local(Path::new(rel).canonicalize().unwrap());
-        assert_eq!(Source::parse(rel).unwrap(), exp);
-        assert_eq!(Source::parse(abs).unwrap(), exp);
-        assert_eq!(Source::parse(abs_url).unwrap(), exp);
+        // Skip checking paths on Windows
+        #[cfg(not(windows))]
```

Comment:
should this be re-written to use [std::path](https://doc.rust-lang.org/std/path/index.html) instead of strings?

Reply:
Realized that the diff was just the weird windows canonicalize thing, so dunce::canonicalize fixes that